### PR TITLE
[BEAM-102] update capability matrix

### PR DESCRIPTION
**Pull request by mxm**
This updates the matrix to the most recent development status.

CC @aljoscha

### CODE REVIEW

**Comment by aljoscha** [comment]
I think it would be good to use more or less the description that we have for the Dataflow runner. This pretty accurately describes the features/limitations.

**Comment by mxm** [comment]
> Batch implemented supports a distributed implementation, but streaming mode may force some size restrictions. Neither mode is able to push lookups directly up into key-based sources. 

Will add this.

File: _data/capability-matrix.yml [hunk 1]
```diff
@@ -120,10 +120,9 @@ categories:
             l2: some size restrictions in streaming
             l3: Batch implemented supports a distributed implementation, but streaming mode may force some size restrictions. Neither mode is able to push lookups directly up into key-based sources.
           - class: flink
-            jira: BEAM-102
-            l1: 'Partially'
-            l2: no supported in streaming
-            l3: Supported in batch. Side inputs for streaming are currently WiP.
+            l1: 'Yes'
+            l2: fully supported
```
Comment (aljoscha) [inline _data/capability-matrix.yml]: could be "some size restrictions in streaming" as for Dataflow, just nitpicking ... 😅 🙊

Comment (mxm) [inline _data/capability-matrix.yml]: That's fine. Thanks!

**Comment by aljoscha** [comment]
LGTM now!

**Comment by mxm** [comment]
Actually there is a typo in the comment above. Will fix before merging.

**Pull request closed without merging**